### DISCUSSION TITLE
Adding custom db engine/connectors

### DISCUSSION THREAD

**Discussion by BobobUnicorn** [prisma/prisma#16206] [Q&A]
As far as I can tell, the only way one could use a custom DB engine/connector is to fork https://github.com/prisma/prisma-engines, add the custom query/introspection/migration engines, and compile it oneself while setting env variables as appropriate. This seems like quite a lot of work and isn't exactly ergonomic. Is there a better way to do this?

**Comment by nurul3101** [comment]
Hello @BobobUnicorn 👋 

Creating a new custom db engine/connector is a considerable effort and would require planning from the product/engineering team. Which db engine/connector do you plan to add? Is there any existing Feature request for it?
And yes, you would need to fork the [prisma-engines](https://github.com/prisma/prisma-engines) repository to make the custom changes. It may also require some changes in the main prisma repository.

**Reply by BobobUnicorn** [reply]
The feature request in the v1 repo: https://github.com/prisma/prisma1/issues/1660
The feature request here: https://github.com/prisma/prisma/issues/8636

I'd asked primarily as a curiosity question about whether or not doing this myself as a proof-of-concept would be feasible; it seems that it's not particularly simple.

**Reply by nurul3101** [reply]
Yes, that's correct! Creating a connector would be a substantial effort involving different teams. 🙂

**Reply by vladi-stevanovic** [reply]
👋 Hi @BobobUnicorn, I'm checking in to see if there's anything else we can clarify in this discussion? 

Let us know if you still have any questions! Or, feel free to post the approach you decided to take and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
